src/libelf-integers.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * (c) 2018 Libelf.JS
+ * Arbitrary integer emulation
+ * Based on https://github.com/pierrec/js-cuint
+ */
+
+var ElfUInt = function (width) {
+    // Configure properties based on bit-width
+    var mask = (1 << (((width - 1) % 16) + 1)) - 1;
+    var nchunks = (width + 15) / 16;
+
+    // Return class
+    return function (value) {
+        this.width = width;
+        this.chunks = new Uint16Array(nchunks);
+        this.mask = mask;
+
+        // Initialization:
+        // Several checks are omitted since JavaScript will cast any undefined/NaN's
+        // into the appropriate type while writing to the chunks array.
+        if (value == null) {
+            this.chunks.fill(0);
+        } else {
+            // Initialize from Number
+            if (typeof value === 'number') {
+                if (!Number.isSafeInteger(value)) {
+                    console.warn(
+                        'Libelf.js: number ' + value +
+                        ' is beyond 53 bits integer precision, use other initialization formats for better precision'
+                    );
+                }
+                if (value > Math.pow(2, this.width)-1 || value < -Math.pow(2, this.width-1)) {
+                    console.warn(
+                        'Libelf.js: number ' + value +
+                        ' overflows ' + this.width + ' bits width, use larger width to keep higher bits'
+                    );
+                }
+                for (var i = 0; i < this.chunks.length; i++) {
+                    // Apply mask on the last chunk to cast to correct width
+                    this.chunks[i] = value & (i === this.chunks.length - 1 ? this.mask : 0xFFFF);
+                    if (value < 0) {
+                        // Fix round off of bits in negative values
+                        value -= 65535
```

Comment:
nit: For the sake of consistency with the rest of the file, this should be 0xFFFF.